src/wp-includes/class-wp-scripts.php

```diff
@@ -289,12 +289,14 @@ public function do_item( $handle, $group = false ) {
 			$ver = $ver ? $ver . '&amp;' . $this->args[ $handle ] : $this->args[ $handle ];
 		}
 
-		$src                   = $obj->src;
-		$strategy              = $this->get_eligible_loading_strategy( $handle );
-		$intended_strategy     = (string) $this->get_data( $handle, 'strategy' );
-		$ie_conditional_prefix = '';
-		$ie_conditional_suffix = '';
-		$conditional           = isset( $obj->extra['conditional'] ) ? $obj->extra['conditional'] : '';
+		$src               = $obj->src;
+		$strategy          = $this->get_eligible_loading_strategy( $handle );
+		$intended_strategy = (string) $this->get_data( $handle, 'strategy' );
+		$conditional       = isset( $obj->extra['conditional'] ) ? true : false;
```

Comment:
Certainly an edge case, but I believe this changes treats a possibly falsy conditional (that was previously treated as no conditional) to be treated as a conditional. Something like this preserves the falsy case:

```suggestion
		$conditional       = (bool) ( $obj->extra['conditional'] ?? false );
```

I don't _think_ this variable is used elsewhere, so I'd probably move the check earlier, above this section, before defining the other variables:

```php
$obj = $this->registered[ $handle ];
if ( $obj->extra['conditional'] ?? false ) {
	return false;
}
```